File: benchmarks/pb_sum_list/mono.py

def f_rcv(ary):
    ct = 0
    for i in ary:
        ct += i
    return ct

def f_call(n):

    a = range(1000)

    v = 0
    for i in xrange(n):
        v += f_rcv(a)

    expect = 499500 * n
    assert v == expect

def run_iter(n):
    f_call(n)

run_iter(900000)

File: benchmarks/pb_return_simple/mono.php
<?php

function f_ret() {
	return 1;
}

function f_call($n) {

	$ct = 0;
	for ($i = 0; $i < $n; $i++) {
		$ct += f_ret();
	}
	assert($ct == $n);
}

function run_iter($n) {
	f_call($n);
}

?>
